middle_end/flambda2/simplify_shared/inlining_helpers.ml

```diff
@@ -32,57 +35,32 @@ let make_inlined_body ~callee ~params ~args ~my_closure ~my_depth ~rec_info
   let body = bind_depth ~my_depth ~rec_info ~body in
   apply_renaming body perm
 
-let wrap_inlined_body_for_exn_support acc ~extra_args ~apply_exn_continuation
+let wrap_inlined_body_for_exn_extra_args acc ~extra_args ~apply_exn_continuation
     ~apply_return_continuation ~result_arity ~make_inlined_body
     ~apply_cont_create ~let_cont_create =
-  (* We need to add a wrapper for the exception handler, so that exceptions
-     coming from the inlined body go through the wrapper and are re-raised with
-     the correct extra arguments.
-
-     This means we also need to add a push trap before the inlined body, and a
-     pop trap after.
-
-     The push trap is simply a matter of jumping to the body, while the pop trap
-     needs to replace the body's return continuation with a wrapper that pops
-     then jumps back. *)
-  (*
-   * As a result, the application [Apply_expr f (args) <k> «k_exn»]
-   * is replaced (before the actual inlining) by:
+  (* We need to add a wrapper for the exception handler so that exceptions
+   * coming from the inlined body are raised with the correct extra arguments:
    *
    * [let_cont_exn k1 (exn: val) =
-   *   Apply_cont k_exn exn extra_args
+   *   (* inlined body here, used to be: Apply_cont k_exn exn extra_args *)
```

Comment:
Is the code actually putting the inlined body there ? That seems wrong.

Reply:
Oh yes, whoops.  Fixed